ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/permissioning/NodeWhitelistController.java

```diff
@@ -180,6 +191,29 @@ public boolean isPermitted(final Peer node) {
     return nodesWhitelist;
   }
 
+  public synchronized void reload() throws RuntimeException {
+    final List<Peer> currentAccountsList = new ArrayList<>(nodesWhitelist);
+    nodesWhitelist.clear();
+
+    try {
+      final PermissioningConfiguration updatedConfig =
+          PermissioningConfigurationBuilder.permissioningConfigurationFromToml(
+              configuration.getConfigurationFilePath(),
+              configuration.isNodeWhitelistEnabled(),
+              configuration.isAccountWhitelistEnabled());
+
+      readNodesFromConfig(updatedConfig);
+      this.configuration = updatedConfig;
+    } catch (Exception e) {
+      LOG.warn(
+          "Error reloading permissions file. In-memory whitelisted nodes will be reverted to previous valid configuration. "
+              + "Details: {}",
+          e.getMessage());
+      this.nodesWhitelist.addAll(currentAccountsList);
```

Comment:
There'd be a chance that reading from config could fail mid-execution, yeah?  Should there be a `nodesWhitelist.clear();` call before the `addAll` ?

Reply:
We read the file and build the list before adding them to the controller's whitelist. When we call `readNodesFromConfig(updatedConfig);` the list has been completely read from the file.

Reply:
I guess we can have the clear() just in case :)